src/plugins/data/public/data_sources/register_default_datasource.ts

```diff
@@ -4,23 +4,32 @@
  */
 
 import { i18n } from '@osd/i18n';
+import { htmlIdGenerator } from '@elastic/eui';
 import { DataPublicPluginStart } from '../types';
-import { DefaultDslDataSource } from './default_datasource';
 
 export const DEFAULT_DATASOURCE_TYPE = 'DEFAULT_INDEX_PATTERNS';
 export const DEFAULT_DATASOURCE_NAME = i18n.translate('data.datasource.type.openSearchDefault', {
   defaultMessage: 'OpenSearch Default',
 });
 
-export const registerDefaultDatasource = (data: Omit<DataPublicPluginStart, 'ui'>) => {
-  // Datasources registrations for index patterns datasource
+export const registerDefaultDataSource = (data: Omit<DataPublicPluginStart, 'ui'>) => {
+  // Registrations of index patterns as default data source
   const { dataSourceService, dataSourceFactory } = data.dataSources;
-  dataSourceFactory.registerDataSourceType(DEFAULT_DATASOURCE_TYPE, DefaultDslDataSource);
   dataSourceService.registerDataSource(
     dataSourceFactory.getDataSourceInstance(DEFAULT_DATASOURCE_TYPE, {
+      id: htmlIdGenerator('local-cluster')('indices'),
       name: DEFAULT_DATASOURCE_NAME,
       type: DEFAULT_DATASOURCE_TYPE,
-      metadata: null,
+      metadata: {
+        ui: {
+          label: 'Index patterns', // display name of your data source,
+          typeLabel: 'Index patterns', // display name of your data source type,
+          typeGroup: 'DEFAULT_INDEX_PATTERNS',
```

Comment:
i think this can be an enum

Reply:
Thanks for early review, thinking about the same, will update it.